src/gameplay/game.ts

```diff
@@ -679,6 +712,52 @@ class Game extends Room {
     }, timeEachLoop);
   }
 
+  StartTimer(timer) {
+    let remainingTimer = timer;
+    
+    const PhaseTimer = setInterval(() => {
+      remainingTimer -= 1;
+      // console.log(remainingTimer);
+
+      for (let i = 0; i < this.playersInGame.length; i++) {
+        const index = usernamesIndexes.getIndexFromUsername(
+          this.socketsOfPlayers,
+          this.playersInGame[i].request.user.username,
+        );
+        // need to go through all sockets, but only send to the socket of players in game
+        if (this.socketsOfPlayers[index]) {
+          this.socketsOfPlayers[index].emit('UpdateTimer', remainingTimer);
+          // console.log("Sent to player: " + this.playersInGame[i].request.user.username + remainingTimer);
+        }
+      }
+
+      let leavePlayers = getAllInactivePlayers(this);
+      console.log('Inactive Players: ' + leavePlayers);
+      if (remainingTimer <= 0 && leavePlayers.length > 0) {
+        let otherPlayers = [];
+        for(const player of this.playersInGame){
+          if(!leavePlayers.includes(player.request.user.username)){
+            otherPlayers.push(player.request.user.username);
+          }
+        }
+        this.voidedGame();
+        punishingPlayers(leavePlayers, otherPlayers);
+        clearInterval(PhaseTimer);
```

Comment:
Double check this works! Not sure if PhaseTimer is defined at this point in time.